okapi-core/src/main/java/org/folio/okapi/managers/DeploymentManager.java

```diff
@@ -181,21 +179,19 @@ public DeploymentManager(Vertx vertx, DiscoveryManager dm, EnvManager em,
       ModuleHandle mh = ModuleHandleFactory.create(vertx, descriptor,
           md1.getSrvcId(), ports, moduleHost, usePort, config);
       Promise<DeploymentDescriptor> promise = Promise.promise();
-      mh.start(future -> {
-        if (future.failed()) {
+      mh.start().onComplete(res -> {
+        if (res.failed()) {
           ports.free(usePort);
           logger.warn("Deploying {} failed", md1.getSrvcId());
-          promise.fail(new OkapiError(ErrorType.USER, future.cause().getMessage()));
+          promise.fail(new OkapiError(ErrorType.USER, res.cause().getMessage()));
           return;
         }
         DeploymentDescriptor md2
             = new DeploymentDescriptor(md1.getInstId(), md1.getSrvcId(),
             moduleUrl, descriptor, mh);
         md2.setNodeId(md1.getNodeId() != null ? md1.getNodeId() : host);
         list.put(md2.getInstId(), md2);
-        dm.add(md2).onComplete(res -> {
-          promise.complete(md2);
-        });
+        dm.add(md2).onComplete(res1 -> promise.handle(res1.map(md2)));
       });
       return promise.future();
     });
```

Comment:
@SuppressWarnings("indentation")  // indentation of fail -> {

...

      return mh.start().compose(started -> {
        DeploymentDescriptor md2 = new DeploymentDescriptor(md1.getInstId(), md1.getSrvcId(),
            moduleUrl, descriptor, mh);
        md2.setNodeId(md1.getNodeId() != null ? md1.getNodeId() : host);
        list.put(md2.getInstId(), md2);
        return dm.add(md2).map(md2);
      }, fail -> {
        ports.free(usePort);
        logger.warn("Deploying {} failed", md1.getSrvcId());
        return Future.failedFuture(new OkapiError(ErrorType.USER, fail.getMessage()));
      });

Reply:
ok